frontend/src/views/Secrets.vue

```diff
@@ -188,21 +149,91 @@ export default {
         speedDial: false
       },
       initialDialogState: {},
-      floatingButton: false
+      defaultSecretTableOptions: {
+        itemsPerPage: 10,
+        sortBy: ['name'],
+        sortDesc: [false]
+      },
+      secretSelectedColumns: {},
+      secretTableOptions: undefined,
+      secretFilter: '',
+      createSecretMenu: false
     }
   },
   computed: {
     ...mapGetters([
       'cloudProfilesByCloudProviderKind',
-      'getInfrastructureSecretByName'
+      'getInfrastructureSecretByName',
+      'infrastructureSecretList',
+      'shootList',
+      'canCreateSecrets',
+      'sortedCloudProviderKindList'
     ]),
     hasCloudProfileForCloudProviderKind () {
       return (kind) => {
         return !isEmpty(this.cloudProfilesByCloudProviderKind(kind))
       }
     },
-    showDisabledCloudProviders () {
-      return !!this.$store.state.cfg.showDisabledCloudProviders
+    secretTableHeaders () {
+      const headers = [
+        {
+          text: 'NAME',
+          align: 'start',
+          value: 'name',
+          sortable: true,
+          defaultSelected: true
+        },
+        {
+          text: 'INFRASTRUCTURE',
+          align: 'start',
+          value: 'infrastructure',
+          sortable: true,
+          defaultSelected: true
+        },
+        {
+          text: 'DETAILS',
+          align: 'start',
+          value: 'details',
+          sortable: false,
+          defaultSelected: true
+        },
+        {
+          text: 'USED BY',
+          align: 'start',
+          value: 'relatedShootCount',
+          defaultSelected: true
+        },
+        {
+          text: 'ACTIONS',
+          align: 'end',
+          value: 'actions',
+          sortable: false,
+          defaultSelected: true
+        }
+      ]
+      return map(headers, header => ({
+        ...header,
+        class: 'nowrap',
+        selected: get(this.secretSelectedColumns, header.value, header.defaultSelected)
+      }))
+    },
+    visibleSecretTableHeaders () {
+      return filter(this.secretTableHeaders, ['selected', true])
+    },
+    secretList () {
+      return map(this.infrastructureSecretList, secret => ({
+        name: secret.metadata.name,
+        isOwnSecret: isOwnSecret(secret),
+        ownernamespace: secret.metadata.secretRef.namespace,
+        infrastructure: `${secret.metadata.cloudProviderKind}${secret.metadata.cloudProfileName}`,
```

Comment:
`infrastructure` acts as key, or? I guess we should add a delimiter, like `--`

Reply:
`infrastructure`is used as value for the _Infrastructure_ column. The value is used for the column sorting by the data table

Reply:
still, to be on the safe side I would add the delimiter, even though a name clash is highly unlikely

Reply:
I do not fully understand what you want to achieve by adding a delimiter. How does this make the value more unique. Other than that, multiple entries with the same value are likely. However, I agree that sorting is not perfect, it would be better to search for `cloudProviderKind` and then for `cloudProfileName` and not combined. However we need to implement a custom search function in that case